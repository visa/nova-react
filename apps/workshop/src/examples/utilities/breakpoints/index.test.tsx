/**
 *              Â© 2025 Visa
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 **/
import { fireEvent, render, screen } from '@testing-library/react';
import { axe } from 'jest-axe';

import metaData from './meta.json';

import { ContainerHideBreakpoints } from './container-hide-breakpoints';
import { HideBreakpoints } from './hide-breakpoints';
import { MediaHideBreakpoints } from './media-hide-breakpoints';

const examples = [
  { Component: HideBreakpoints, title: metaData['hide-breakpoints'].title },
  { Component: ContainerHideBreakpoints, title: metaData['container-hide-breakpoints'].title },
  { Component: MediaHideBreakpoints, title: metaData['media-hide-breakpoints'].title },
];

describe('Breakpoints examples', () => {
  examples.forEach(({ Component, title }) => {
    it(`${title} should render correctly`, async () => {
      const { container } = render(<Component />);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
      expect(container).toMatchSnapshot();
    });
  });
  describe('hide', () => {
    it('should hide on hide click', () => {
      render(<HideBreakpoints />);
      const button = screen.getByText('Hide');
      fireEvent.click(button);
      expect(screen.queryByText('Show')).not.toBeNull();
    });
  });
});
