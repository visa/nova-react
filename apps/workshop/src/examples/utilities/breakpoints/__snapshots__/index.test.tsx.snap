// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breakpoints examples Hide should render correctly 1`] = `
<div>
  <div
    class="v-align-items-center v-flex v-gap-24"
  >
    <span
      class="v-avatar"
    >
      AM
    </span>
    <span
      aria-live="polite"
      class="v-sr"
      role="alert"
    />
    <button
      class="v-button v-button-secondary"
    >
      Hide
    </button>
  </div>
</div>
`;

exports[`Breakpoints examples Hide with breakpoint-matching container query should render correctly 1`] = `
<div>
  <div
    class="v-flex v-flex-row v-flex-wrap v-gap-8"
  >
    <span
      class="v-avatar v-xs-container-hide"
      style="--v-avatar-background: #e50000;"
    >
      XS
    </span>
    <span
      class="v-avatar v-sm-container-hide"
      style="--v-avatar-background: orange; --v-avatar-foreground: black;"
    >
      SM
    </span>
    <span
      class="v-avatar v-md-container-hide"
      style="--v-avatar-background: yellow; --v-avatar-foreground: black;"
    >
      MD
    </span>
    <span
      class="v-avatar v-lg-container-hide"
      style="--v-avatar-background: green;"
    >
      LG
    </span>
    <span
      class="v-avatar v-xl-container-hide"
      style="--v-avatar-background: blue;"
    >
      XL
    </span>
    <span
      class="v-avatar v-xxl-container-hide"
      style="--v-avatar-background: purple;"
    >
      XXL
    </span>
    <span
      class="v-avatar v-mobile-container-hide"
    >
      <svg
        aria-hidden="true"
        aria-label="Showing a laptop to indicate we're on a desktop-sized container"
        class="v-icon v-icon-visa v-icon-low v-icon-device-laptop"
        focusable="false"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="v-icon-secondary"
          d="M24 19.0005H0V21.0005H24V19.0005Z"
        />
        <path
          class="v-icon-primary"
          d="M22 17.0005H2V5.00049C2 3.90049 2.9 3.00049 4 3.00049H20C21.1 3.00049 22 3.90049 22 5.00049V17.0005ZM4 15.0005H20V5.00049H4V15.0005Z"
        />
      </svg>
    </span>
    <span
      class="v-avatar v-desktop-container-hide"
    >
      <svg
        aria-hidden="true"
        aria-label="Showing a mobile device to indicate we're on a mobile-sized container"
        class="v-icon v-icon-visa v-icon-low v-icon-device-mobile"
        focusable="false"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="v-icon-secondary"
          d="M15 4H9V6H15V4Z"
        />
        <path
          class="v-icon-primary"
          clip-rule="evenodd"
          d="M5 3C5 1.3457 6.3457 0 8 0H16C17.6543 0 19 1.3457 19 3V21C19 22.6543 17.6543 24 16 24H8C6.3457 24 5 22.6543 5 21V3ZM16 22C16.5518 22 17 21.5518 17 21V20H7V21C7 21.5518 7.44873 22 8 22H16ZM8 2C7.44873 2 7 2.44873 7 3V18H17V3C17 2.44873 16.5518 2 16 2H8Z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`Breakpoints examples Hide with breakpoint-matching media query should render correctly 1`] = `
<div>
  <div
    class="v-flex v-flex-row v-flex-wrap v-gap-8"
  >
    <span
      class="v-avatar v-xs-media-hide"
      style="--v-avatar-background: #e50000;"
    >
      XS
    </span>
    <span
      class="v-avatar v-sm-media-hide"
      style="--v-avatar-background: orange; --v-avatar-foreground: black;"
    >
      SM
    </span>
    <span
      class="v-avatar v-md-media-hide"
      style="--v-avatar-background: yellow; --v-avatar-foreground: black;"
    >
      MD
    </span>
    <span
      class="v-avatar v-lg-media-hide"
      style="--v-avatar-background: green;"
    >
      LG
    </span>
    <span
      class="v-avatar v-xl-media-hide"
      style="--v-avatar-background: blue;"
    >
      XL
    </span>
    <span
      class="v-avatar v-xxl-media-hide"
      style="--v-avatar-background: purple;"
    >
      XXL
    </span>
    <span
      class="v-avatar v-mobile-media-hide"
    >
      <svg
        aria-hidden="true"
        aria-label="Showing a laptop to indicate we're on a desktop-sized media"
        class="v-icon v-icon-visa v-icon-low v-icon-device-laptop"
        focusable="false"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="v-icon-secondary"
          d="M24 19.0005H0V21.0005H24V19.0005Z"
        />
        <path
          class="v-icon-primary"
          d="M22 17.0005H2V5.00049C2 3.90049 2.9 3.00049 4 3.00049H20C21.1 3.00049 22 3.90049 22 5.00049V17.0005ZM4 15.0005H20V5.00049H4V15.0005Z"
        />
      </svg>
    </span>
    <span
      class="v-avatar v-desktop-media-hide"
    >
      <svg
        aria-hidden="true"
        aria-label="Showing a mobile device to indicate we're on a mobile-sized media"
        class="v-icon v-icon-visa v-icon-low v-icon-device-mobile"
        focusable="false"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="v-icon-secondary"
          d="M15 4H9V6H15V4Z"
        />
        <path
          class="v-icon-primary"
          clip-rule="evenodd"
          d="M5 3C5 1.3457 6.3457 0 8 0H16C17.6543 0 19 1.3457 19 3V21C19 22.6543 17.6543 24 16 24H8C6.3457 24 5 22.6543 5 21V3ZM16 22C16.5518 22 17 21.5518 17 21V20H7V21C7 21.5518 7.44873 22 8 22H16ZM8 2C7.44873 2 7 2.44873 7 3V18H17V3C17 2.44873 16.5518 2 16 2H8Z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;
